pkg/plugins/runtime/k8s/controllers/gatewayapi/gateway_controller.go

```diff
@@ -181,6 +181,14 @@ func gatewaysForRoute(l logr.Logger) kube_handler.MapFunc {
 
 		var requests []kube_reconcile.Request
 		for _, parentRef := range route.Spec.ParentRefs {
+			if parentRef.Group != nil && *parentRef.Group != gatewayapi.GroupName {
+				continue
+			}
+
+			if parentRef.Kind != nil && *parentRef.Kind != "Gateway" {
```

Comment:
These are always `!= nil` because of defaulting so we can just:

```suggestion
			if *parentRef.Group != gatewayapi.GroupName || *parentRef.Kind != "Gateway" {
```

Reply:
Good catch!